crates/typst-library/src/meta/provide.rs

```diff
@@ -0,0 +1,43 @@
+use crate::prelude::*;
+
+/// Provide custom metadata for the Typst query system.
+///
+/// The Typst query system allows users to extract metadata from the document, offering both a
+/// generic selector and a specific key-value mechanism. The 'provide' command is an essential
+/// part of this mechanism, associating a single piece of metadata with the specified key.
+/// Subsequent invocations will append to a list of values for the same key.
```

Comment:
`provide` could also be useful in combination with the `query()` function when you want to leave some arbitrary value in the document (which isn't really possible at the moment). Maybe we could explain that here, too?

Reply:
I've made a short hint. But "state" should already be good to plant and retrieve arbitrary values, right? Should we be able to query the final versions of state, too?

Reply:
expose could be more convenient than state to emulate user-defined custom elements. because it works just like elements do.